substrate/frame/parameters/src/tests/mock.rs

```diff
@@ -0,0 +1,139 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg(any(test, feature = "runtime-benchmarks"))]
+
+//! Mock runtime that configures the `pallet_example_basic` to use dynamic params for testing.
+
+use frame_support::{
+	construct_runtime, derive_impl,
+	dynamic_params::{dynamic_pallet_params, dynamic_params},
+	parameter_types,
+	traits::EnsureOriginWithArg,
+};
+
+use crate as pallet_parameters;
+use crate::*;
+
+#[derive_impl(frame_system::config_preludes::TestDefaultConfig as frame_system::DefaultConfig)]
+impl frame_system::Config for Runtime {
+	type Block = frame_system::mocking::MockBlock<Runtime>;
+	type AccountData = pallet_balances::AccountData<<Self as pallet_balances::Config>::Balance>;
+}
+
+#[derive_impl(pallet_balances::config_preludes::TestDefaultConfig as pallet_balances::DefaultConfig)]
+impl pallet_balances::Config for Runtime {
+	type ReserveIdentifier = [u8; 8];
+	type AccountStore = System;
+}
+
+#[docify::export]
+#[dynamic_params(RuntimeParameters, pallet_parameters::Parameters::<Runtime>)]
+pub mod dynamic_params {
+	use super::*;
+
+	#[dynamic_pallet_params]
+	#[codec(index = 3)]
+	pub mod pallet1 {
+		#[codec(index = 0)]
+		pub static Key1: u64 = 0;
```

Comment:
would it be simpler to not have inner levels as `pallet1` and `pallet2`? just `dynamic_params::FellowshipSalaryBudget` for example

Reply:
But then you are writing `dynamic_params::FellowshipSalaryBudget` instead of `dynamic_params::fellowship::SalaryBudget`.  
It would be simpler - yes, but i think grouping them by topic can be useful.

Reply:
I would go for simpler solution, to me namespaces do not look so important in this case. Prefix for a name always can be applied. Our regular parameters also do not have namespaces.